wgpu-types/src/lib.rs

```diff
@@ -2318,6 +2327,14 @@ pub enum TextureFormat {
     /// [`Features::DEPTH32FLOAT_STENCIL8`] must be enabled to use this texture format.
     Depth32FloatStencil8,
 
+    /// YUV 4:2:0 video resource format.
+    /// Valid luminance data view formats for this video resource format are [`TextureFormat::R8Unorm`] and [`TextureFormat::R8Uint`].
+    /// Valid chrominance data view formats (width and height are each 1/2 of luminance view) for this video resource format are [`TextureFormat::Rg8Unorm`] and [`TextureFormat::Rg8Uint`].
+    /// Width and height must be even.
```

Comment:
There doesn't appear to be any validation of this, unless I'm missing something.

We should add a test for this (sorry I didn't catch this the first round).

One way this could be implemented is if we treat NV12's block size as (2, 2), then it will use the compressed texture block size validation.